src/components/Modal/__tests__/index.tsx

```diff
@@ -46,6 +46,19 @@ describe('Modal', () => {
         <div>modal</div>
       </Modal>,
     ))
+
+  test(`renders with disclosure and onBeforeClose`, () =>
+    shouldMatchEmotionSnapshotWithPortal(
+      <Modal
+        ariaLabel="modal-test"
+        baseId="modal-test"
+        disclosure={() => <button type="button">Test</button>}
+        onBeforeClose={() => console.log('onBeforeClose triggered')}
```

Comment:
mock a jest function and count how many time the function is involved ?

Reply:
Good idea 🔝